ethcore/sync/src/chain/handler.rs

```diff
@@ -673,8 +673,7 @@ impl SyncHandler {
 		trace!(target: "sync", "{:02} -> Transactions ({} entries)", peer_id, item_count);
 		let mut transactions = Vec::with_capacity(item_count);
 		for i in 0 .. item_count {
-			let rlp = r.at(i)?;
-			let tx = rlp.as_raw().to_vec();
+			let tx = r.val_at(i)?;
```

Comment:
The whole point of passing `Bytes` was to avoid doing any intensive work (like hashing transactions) synchronously in the network `IoHandlers`.
The whole refactoring was done after a careful profiling of issues that happens on the mainnet where `network` lock could be acquired for a very long time.